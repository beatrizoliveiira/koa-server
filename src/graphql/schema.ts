import { buildSchema } from 'graphql';

export default buildSchema(`
  type Event {
    id: ID
    title: String
    allDay: Boolean
    start: String
    end: String
  }
  type Query {
    listEvents: [Event]
  }
  type Mutation {
    createEvent(
      title: String!,
      start: String, 
      end: String, 
      allDay: Boolean): Event
  }
`);