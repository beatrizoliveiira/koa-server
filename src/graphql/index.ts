export { registerLoader, getDataloaders } from "./loaderRegister";
export { connectionArgs, connectionDefinitions } from "./connectionDefinitions";