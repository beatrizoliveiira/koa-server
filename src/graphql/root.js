const Event = require('../models/event');

module.exports = {
  listEvents() {
    return Event.find();
  },
  createEvent(input) {
    const newEvent = new Event(input);
    return newEvent.save();
  },
}