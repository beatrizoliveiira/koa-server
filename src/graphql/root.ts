import Event from '../models/event';

export function listEvents() {
  return Event.find();
}
export function createEvent(input) {
  const newEvent = new Event(input);
  return newEvent.save();
}