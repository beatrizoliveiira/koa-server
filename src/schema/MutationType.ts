import { GraphQLObjectType } from 'graphql';

import EventMutations from '../modules/event/mutations';

const MutationType = new GraphQLObjectType({
  name: 'Mutation',
  fields: () => ({
  ...EventMutations,
  }),
});

export default MutationType;