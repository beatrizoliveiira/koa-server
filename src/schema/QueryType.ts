import { GraphQLObjectType, GraphQLNonNull } from 'graphql';

import { connectionArgs } from "../graphql/connectionDefinitions";

import { nodesField, nodeField } from '../modules/node/typeRegister';
import * as EventLoader from '../modules/event/EventLoader';
import { EventConnection } from '../modules/event/EventType';

const QueryType = new GraphQLObjectType({
  name: 'Query',
  description: 'The root of all... queries',
  fields: () => ({
    node: nodeField,
    nodes: nodesField,
    events: {
      type: new GraphQLNonNull(EventConnection.connectionType),
      args: {
        ...connectionArgs,
      },
      resolve: async (_, args, context) => await EventLoader.loadAll(context, args),
    },
  }),
});

export default QueryType;
