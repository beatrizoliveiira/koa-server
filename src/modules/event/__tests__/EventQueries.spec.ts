import { graphql } from "graphql";

import {
  clearDbAndRestartCounters,
  connectMongoose,
  disconnectMongoose,
  sanitizeTestObject,
} from "../../../../test";

import { schema } from "../../../schema/schema";

import { createEvent } from "../fixture/createEvent";

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

it("should query all events", async () => {
  const event = await createEvent({
    name: "Event",
    start: "2019-01-01T00:00:00.000Z",
    end: "2019-01-01T23:59:59.000Z",
    allDay: true,
  });

  const anotherEvent = await createEvent({
    name: "Another Event",
    start: "2019-01-02T00:00:00.000Z",
    end: "2019-01-02T23:59:59.000Z",
    allDay: false,
  });

  // language=GraphQL
  const query = `
    query Q {
      events(first: 10) {
        edges {
          node {
            name
          }
        }
      }
    }
  `;

  const rootValue = {};
  //const contextValue = await getContext({});
  const variableValues = {};

  const result = await graphql({
    schema,
    source: query,
    rootValue,
    //contextValue,
    variableValues,
  });

  expect(result.errors).toBeUndefined();

  // eslint-disable-next-line
  console.log("result: ", result.data.events);

  expect(result.data.events.edges.length).toBe(2);

  expect(result.data.events.edges[0].node.name).toBe(event.name);
  expect(result.data.events.edges[1].node.name).toBe(anotherEvent.name);

  expect(sanitizeTestObject(result.data)).toMatchSnapshot();
});
