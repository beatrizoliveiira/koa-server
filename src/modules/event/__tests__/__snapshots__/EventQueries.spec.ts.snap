// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should query all events 1`] = `
Object {
  "events": Object {
    "edges": Array [
      Object {
        "node": Object {
          "allDay": false,
          "end": "2019-01-02T23:59:59.000Z",
          "name": "event B",
          "start": "2019-01-02T00:00:00.000Z",
        },
      },
      Object {
        "node": Object {
          "allDay": true,
          "end": "2019-01-01T23:59:59.000Z",
          "name": "event A",
          "start": "2019-01-01T00:00:00.000Z",
        },
      },
    ],
  },
}
`;
